comptests/prov/test_prov_k8s_cortx_upgrade.py

```diff
@@ -269,14 +268,82 @@ def test_32605(self):
         LOGGER.info("Step 5: Done.")
 
         LOGGER.info("Step 6: Test parallel upgrades.")
+        que = queue.Queue()
         parallel_upgrade_message = "An upgrade is already in progress"
-        upgrade_thread = Thread(target=self.perform_upgrade)
-        parallel_upgrade_thread = Thread(target=self.perform_upgrade, args=(False,))
+        upgrade_thread = Thread(target=self.perform_upgrade, args=(True, que,))
+        parallel_upgrade_thread = Thread(target=self.perform_upgrade, args=(False, que,))
         upgrade_thread.start()
         time.sleep(10)  # Wait to start upgrade_thread thread
         parallel_upgrade_thread.start()
-        resp = self.que.get(1)
+        resp = que.get(1)
         assert_utils.assert_in(parallel_upgrade_message, resp[1])
         upgrade_thread.join()
         LOGGER.info("Step 6: Done.")
         LOGGER.info("Test Completed.")
+
+    @pytest.mark.lc
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37514")
+    def test_37514(self):
+        """
+        Verify Upgrade script throws an error message when invalid argument is passed.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when invalid argument passed to upgrade script.")
+        cmd = commands.UPGRADE_NEG_CMD.format(self.prov_deploy_cfg["git_remote_path"]) + " -abc"
+        resp = self.master_node_obj.execute_cmd(cmd=cmd, exc=False)
+        if isinstance(resp, bytes):
+            resp = str(resp, 'UTF-8')
+        resp = "".join(resp).replace("\\n", "\n")
+        assert_utils.assert_in("Invalid argument provided", resp)
+        LOGGER.info("Test Completed.")
+
+    @pytest.mark.lc
+    @pytest.mark.comp_prov
+    @pytest.mark.tags("TEST-37496")
+    def test_37496(self):
+        """
+        Verify Upgrade script throws an error message when no argument is passed.
+        """
+        LOGGER.info("Test Started.")
+        LOGGER.info("Check proper error message when no argument passed to upgrade script.")
+        error_msg = "ERROR: Required option POD_TYPE is missing."
+        cmd = commands.UPGRADE_NEG_CMD.format(self.prov_deploy_cfg["git_remote_path"])
+        resp = self.master_node_obj.execute_cmd(cmd=cmd, exc=False)
+        if isinstance(resp, bytes):
+            resp = str(resp, 'UTF-8')
+        resp = "".join(resp).replace("\\n", "\n")
+        print(resp)
```

Comment:
Can you replace print with log statement ?

Reply:
Removed. Actually this is not required.